airbyte/mcp/cloud_ops.py

```diff
@@ -690,116 +707,18 @@ def register_cloud_ops_tools(app: FastMCP) -> None:
     """@private Register tools with the FastMCP app.
 
     This is an internal function and should not be called directly.
-    """
-    app.tool(
-        check_airbyte_cloud_workspace,
-        annotations={
-            READ_ONLY_HINT: True,
-            IDEMPOTENT_HINT: True,
-        },
-    )
-
-    app.tool(
-        deploy_source_to_cloud,
-        annotations={
-            DESTRUCTIVE_HINT: False,
-        },
-    )
-
-    app.tool(
-        deploy_destination_to_cloud,
-        annotations={
-            DESTRUCTIVE_HINT: False,
-        },
-    )
-
-    app.tool(
-        deploy_noop_destination_to_cloud,
-        annotations={
-            DESTRUCTIVE_HINT: False,
-        },
-    )
-
-    app.tool(
-        create_connection_on_cloud,
-        annotations={
-            DESTRUCTIVE_HINT: False,
-        },
-    )
-
-    app.tool(
-        run_cloud_sync,
-        annotations={
-            DESTRUCTIVE_HINT: False,
-        },
-    )
-
-    app.tool(
-        get_cloud_sync_status,
-        annotations={
-            READ_ONLY_HINT: True,
-            IDEMPOTENT_HINT: True,
-        },
-    )
 
-    app.tool(
-        get_cloud_sync_logs,
-        annotations={
-            READ_ONLY_HINT: True,
-            IDEMPOTENT_HINT: True,
-        },
-    )
-
-    app.tool(
-        list_deployed_cloud_source_connectors,
-        annotations={
-            READ_ONLY_HINT: True,
-            IDEMPOTENT_HINT: True,
-        },
-    )
-
-    app.tool(
-        list_deployed_cloud_destination_connectors,
-        annotations={
-            READ_ONLY_HINT: True,
-            IDEMPOTENT_HINT: True,
-        },
-    )
-
-    app.tool(
-        list_deployed_cloud_connections,
-        annotations={
-            READ_ONLY_HINT: True,
-            IDEMPOTENT_HINT: True,
-        },
-    )
-
-    app.tool(
-        publish_custom_source_definition,
-        annotations={
-            DESTRUCTIVE_HINT: False,
-        },
-    )
+    Tools are filtered based on safe mode settings:
+    - AIRBYTE_CLOUD_MCP_READONLY_MODE=1: Only read-only tools are registered
+    - AIRBYTE_CLOUD_MCP_SAFE_MODE=1: Destructive tools are not registered
+    """
+    current_module = sys.modules[__name__]
 
-    app.tool(
-        list_custom_source_definitions,
-        annotations={
-            READ_ONLY_HINT: True,
-            IDEMPOTENT_HINT: True,
-        },
-    )
+    for name in dir(current_module):
```

Comment:
The "dir(current_module)" approach is messy. Update the cloud_tool() implementation to maintain its own list of registered tools so that you can then iterate over them.